packages/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift

```diff
@@ -12,25 +12,14 @@ public class ExpoReactDelegate: NSObject {
   }
 
   @objc
-  public func createBridge(delegate: RCTBridgeDelegate, launchOptions: [AnyHashable: Any]?) -> RCTBridge {
-    self.handlers.forEach { $0.bridgeWillCreate() }
-    let result = self.handlers.lazy
-      .compactMap { $0.createBridge(reactDelegate: self, bridgeDelegate: delegate, launchOptions: launchOptions) }
-      .first(where: { _ in true }) ?? RCTBridge(delegate: delegate, launchOptions: launchOptions)!
-    self.handlers.forEach { $0.bridgeDidCreate(bridge: result) }
-    return result
-  }
-
-  @objc
-  public func createRootView(
-    bridge: RCTBridge,
+  public func createReactRootView(
     moduleName: String,
     initialProperties: [AnyHashable: Any]?,
-    fabricEnabled: Bool = EXAppDefines.APP_NEW_ARCH_ENABLED
+    launchOptions: [UIApplication.LaunchOptionsKey: Any]?
   ) -> UIView {
     return self.handlers.lazy
-      .compactMap { $0.createRootView(reactDelegate: self, bridge: bridge, moduleName: moduleName, initialProperties: initialProperties) }
-      .first(where: { _ in true }) ?? EXAppSetupDefaultRootView(bridge, moduleName, initialProperties, fabricEnabled)
+      .compactMap { $0.createReactRootView(reactDelegate: self, moduleName: moduleName, initialProperties: initialProperties, launchOptions: launchOptions) }
+      .first(where: { _ in true }) ?? ExpoReactRootViewFactory.createDefaultReactRootView(nil, moduleName: moduleName, initialProperties: initialProperties)
```

Comment:
my understanding is that `.first(where: { _ in true })` will loop all handlers until all handlers return nil and try createDefaultReactRootView. however, `.first` just gets the first handler. if the first handler is nil, it will try createDefaultReactRootView.

please correct me if my understanding is wrong.

Reply:
Doesn't `{ _ in true }` only evaluate for the first element? I would say it's equivalent to `.first`, but I'm not fully sure though 😅

Reply:
tested that and it has slightly difference. with `where: { _ in true }` that's more expected without looping the handler again.
also a reference from https://github.com/apple/swift/issues/55374

Reply:
Ok, thanks for clarifying!